### DISCUSSION TITLE
ws-m181 gps is detected properly but cannot get a fix

### DISCUSSION THREAD

**Discussion by sgrizzi** [iNavFlight/inav#9258] [Q&A]
I am using a ws-m181 gps connected to a CrazyF405 FC, running INAV 6.1 firmware. The FC is powered with a 1s battery.
All sensors are detected correctly, including the GPS (icons are all blue).
GPS is on uart1 @115200 baud. Protocol UBLOX7 (I also tried UBLOX with no changes). Galileo satellites also active.
Update time: ca. 10Hz. "Total messages" count up steadily.

With this configuration I get a fix randomly, very seldom...
When I use gpspassthough in CLI I can see only garbled text... rarely do I see output strings in clear.
Connections are correct -and sometimes the gps does work...
The board is powered with a LiPo @3.9V.

Has anybody any hint for me on how to fix this problem? is there a way (e.g. a CLI command) to force a restart/re-initialization of uart1?
Thanks in advance.

**Comment by capblack** [comment]
hello the voltage is not enough for the correct operation of the gps, it needs 5v minimum 4.5v but not recommended and use uart in auto no 115200

**Comment by sgrizzi** [comment]
That was my fear... but an onboard buck-boost is generating 5V directly from the LiPo.
I measure ca.4.5V when the FC is just supplied via the usb cable, and 4.957V when the LiPo is connected.
The buck-boost takes a bit of time to get to the nominal output voltage, maybe the GPS is initialized before the nominal voltage has been reached.
That was also the reason of my question on whether it is possible to manually re-initialize the GPS.
Unfortunately there is no "auto" selection available for the GPS interface speed.

**Comment by capblack** [comment]
measuring with a multimeter many times is not enough, test with an oscilloscope to verify that you do not have ripples in the 5v and if so add capacitance, if your gps and the FC receive constant 5v, it should work fine.
FC and GPS usually incorporate an r/c that generates a short delay, so they won't be affected until vcc is ok

**Comment by sgrizzi** [comment]
Thanks for the hints. I checked the GPS supply voltage with a scope and it is indeed stable and clean, just below 5V.
Then I checked the serial output data directly at the GPS pins. They looked very clean, swinging between 0 and 3.3V, "consistent" I would say, i.e. the packets start always in the same way. Data look somehow binary "object" format, no clear ascii data, with some bytes probably showing incrementing counters (or time?).
So, in summary, everything looks "normal"... Probably the garbled message via gpspassthrough is related to the fact that the GPS is instructed to talk ublox7, right?
Still I cannot get a fix.
I tried to use NMEA as a protocol, despite of the fact that INAV is warning me "in red" that this will be deprecated.
Well, with NMEA the communication is again in clear, and I did get a fix after ca. 70K messages. Refresh rate is between 10Hz and 12Hz, not bad I would say.
Shall I just stick to NMEA protocol? Ublox packets look way more compact, so I guess that that protocol should be more efficient.
Are there any additional test I can carry out to get Ublox working?

**Reply by capblack** [reply]
hello well I have 3 of those working without any problem and with 12-18 sat in view all the time, but they are connected with F7 and H7 controllers, select only UBLOX and check again the connections that TX/RX are connected to a hard com and not to a software com port. I'll take a look later because the 10hz refresh rate seems strange to me, this gps should only work at 5hz

**Reply by capblack** [reply]
You can also connect this gps to the pc to verify that it is not damaged through an rs232 to usb adapter and you have a lot of software available including UBLOX center

**Reply by capblack** [reply]
inav only work at 5hz for this gps

**Comment by sgrizzi** [comment]
Apologies. It was my fault... actually lack of patience and of expertise :-(
The device was brand new and I did not take "time to first fix" sufficiently into account, nor that it takes several minutes to download all sat data, anyhow. I had read that the WS-M181 was ultra-fast...
Using NMEA protocol in clear did prove to me that the device was connected correctly and I could get a fix with 6 satellites.
Then I moved to UBLOX7 and could get the same fix - though getting 6 "stable" sats was not easy.
Since I live very close to the mountains and I was testing the GPS just outside my house, the sky view was also limited.
Yesterday, from a position with a reasonably wide view of the sky, I could get 8 stable satellites within few minutes - and I let it run for 20 minutes, so I hope it had a chance to load all necessary sat data.
By the way, the refresh rate is now 5Hz (but it definitely was 10-12.5Hz when using NMEA, weird)
Thanks for all your good hints which helped me a lot.
